### PR TITLE
kernel: bump 5.4 to 5.4.58

### PR DESCRIPTION
**Pull request by graysky2**
This PR is a blend of several kernel bumps authored by ldir taken from his
staging tree w/ some further adjustments made by me and `update_kernel.sh`

Summary:
Deleted upstreamed patches:
```
 022-v5.8-mtd-rawnand-brcmnand-correctly-verify-erased-pages.patch
 024-v5.8-mtd-rawnand-brcmnand-fix-CS0-layout.patch
 742-v5.5-net-sfp-add-support-for-module-quirks.patch
 743-v5.5-net-sfp-add-some-quirks-for-GPON-modules.patch
 641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch
 0402-net-ethernet-mtk_eth_soc-Always-call-mtk_gmac0_rgmii.patch
```
Manually merged patches:
```
 395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch
 701-net-0231-enetc-Use-DT-protocol-information-to-set-up-the-port.patch
 950-0132-lan78xx-Debounce-link-events-to-minimize-poll-storm.patch
```
Remaining patches auto-adjusted by update_kernel.sh

Build system: x86_64
Run-tested: Netgear R7800 (ipq806x)

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

### CODE REVIEW

File: target/linux/generic/hack-5.4/641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch [hunk 1]
```diff
@@ -1,114 +0,0 @@
-From a00590d570212c3c633bd463cef8ec7377cc7993 Mon Sep 17 00:00:00 2001
-From: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>
-Date: Tue, 30 Jun 2020 12:07:44 +0100
-Subject: [PATCH] sch_cake: fix IP protocol handling in the presence of VLAN
```
Comment (hauke) [inline target/linux/generic/hack-5.4/641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch:4]: Why do you remove this patch?

**Review by hauke — COMMENTED** [review]
@ldir-EDB0  can you please have a look at the cake changes.

File: target/linux/generic/hack-5.4/641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch (hunk 1 above)
Comment (graysky2) [reply target/linux/generic/hack-5.4/641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch:4]: @hauke - https://git.openwrt.org/?p=openwrt/staging/ldir.git;a=commit;h=35e23f471431f791ac4314c591125aad5af8f56e

File: target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch [hunk 2]
```diff
@@ -1,170 +0,0 @@
-From b0c19ed6088ab41dd2a727b60594b7297c15d6ce Mon Sep 17 00:00:00 2001
-From: =?UTF-8?q?Toke=20H=C3=B8iland-J=C3=B8rgensen?= <toke@redhat.com>
-Date: Fri, 29 May 2020 14:43:44 +0200
-Subject: [PATCH] sch_cake: Take advantage of skb->hash where appropriate
```
[the remark this answers is not in the document]
Comment (graysky2) [reply target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch:4]: @hauke - Initially, `update_kernel.sh` flagged.  My testing procedure is manually apply the patch to a freshly untarred source to confirm.  In this case, patching the source gave a message like, "Reversed (or previously applied) patch detected!  Skipping patch" as I recall.

1) Did I make a mistake?
2) Is there a better/more reliable method to verify that the target patch has been upstreamed?

Thanks.

**Comment by graysky2** [comment]
> Conflicting files
include/kernel-version.mk

Rebased and pushed.

File: target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (ldir-EDB0) [reply target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch:4]: It was backported by upstream

File: target/linux/generic/hack-5.4/641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch (hunk 1 above)
Comment (ldir-EDB0) [reply target/linux/generic/hack-5.4/641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch:4]: It was fixed more generically upstream so no longer required

**Comment by ldir-EDB0** [comment]
I'm really busy with other stuff at the moment - a cursory look over looks good to me.  The cake/qdisc/skb_protocol stuff is fun 'cos upstream backported/fixed in a more global way

**Comment by lucize** [comment]
run tested in mvebu

Tested-By Lucian Cristian <Lucian.cristian@gmail.com>

**Comment by curtdept** [comment]
@graysky2  Not sure how that relates? Reading both threads you seem to have that issue on kernels ranging from .52 -.58, so doesn't really seem like a regression relative to a .58 PR

**Comment by rmandrad** [comment]
Any plans for when this is going to be released ? thank you

**Comment by curtdept** [comment]
run tested in x86/64

Tested-By Curtis Deptuck curtdept@me.com

**Comment by nouman8** [comment]
looks like devs are all busy prepping for linux 5.9 kernel :)

**Comment by adschm** [comment]
I was just considering to pick this, but the list in the description does not match the actual changes, so this means extra work and will take longer ...

File: target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (adschm) [reply target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch:4]: I don't find these changes anywhere in 5.4.58 kernel.

**Comment by graysky2** [comment]
> I was just considering to pick this, but the list in the description does not match the actual changes, so this means extra work and will take longer

@adschm - If you identified a typo or mistake please call it out so I can verify and fix it.

File: target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (graysky2) [reply target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch:4]: > I was just considering to pick this, but the list in the description does not match the actual changes, so this means extra work and will take longer ...

Is the above comment what you meant when you said that the description doesn't match the actual changes?

[the remark this answers is not in the document]
Comment (adschm) [reply target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch:4]: Wait a few minutes, I'm just refreshing myself, will push the result to my staging tree when I'm done.

**Comment by adschm** [comment]
My updated patch is found here:
https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/k54

Changes overview:
- do not drop 395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch, but just adjust to upstream changes
- really fix 950-0132-lan78xx-Debounce-link-events-to-minimize-poll-storm.patch
- actually delete 0402-net-ethernet-mtk_eth_soc-Always-call-mtk_gmac0_rgmii.patch
- update commit message to follow the actual patch actions

**Comment by graysky2** [comment]
@adschm - I will pull. build, and test on my hardware.  Is your proposal for me to just rebase and force push to this PR or did you intent to just request the merge from your branch?  Thanks.

**Comment by graysky2** [comment]
@adschm - When I apply your commit to HEAD (currently 2cac31223d97fffacb8154f5739c67dfe078106f), I get multiple rejects:
```
patching file include/kernel-version.mk
patching file target/linux/apm821xx/patches-5.4/801-usb-xhci-add-firmware-loader-for-uPD720201-and-uPD72.patch
patching file target/linux/apm821xx/patches-5.4/802-usb-xhci-force-msi-renesas-xhci.patch
patching file target/linux/ath79/patches-5.4/0011-MIPS-ath79-select-the-PINCTRL-subsystem.patch
patching file target/linux/ath79/patches-5.4/0032-MIPS-ath79-sanitize-symbols.patch
patching file target/linux/ath79/patches-5.4/910-unaligned_access_hacks.patch
patching file target/linux/bcm27xx/patches-5.4/950-0030-lan78xx-Enable-LEDs-and-auto-negotiation.patch
patching file target/linux/bcm27xx/patches-5.4/950-0111-lan78xx-Read-initial-EEE-status-from-DT.patch
patching file target/linux/bcm27xx/patches-5.4/950-0117-net-lan78xx-Disable-TCP-Segmentation-Offload-TSO.patch
patching file target/linux/bcm27xx/patches-5.4/950-0118-lan78xx-Move-enabling-of-EEE-into-PHY-init-code.patch
patching file target/linux/bcm27xx/patches-5.4/950-0132-lan78xx-Debounce-link-events-to-minimize-poll-storm.patch
patching file target/linux/bcm27xx/patches-5.4/950-0138-lan78xx-EEE-support-is-now-a-PHY-property.patch
patching file target/linux/bcm27xx/patches-5.4/950-0146-lan78xx-use-default-alignment-for-rx-buffers.patch
patching file target/linux/bcm27xx/patches-5.4/950-0214-usb-xhci-Show-that-the-VIA-VL805-supports-LPM.patch
patching file target/linux/bcm27xx/patches-5.4/950-0267-xhci-add-quirk-for-host-controllers-that-don-t-updat.patch
patching file target/linux/bcm27xx/patches-5.4/950-0283-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
patching file target/linux/bcm53xx/patches-5.4/905-BCM53573-minor-hacks.patch
patching file target/linux/bcm63xx/patches-5.4/021-v5.8-mtd-rawnand-brcmnand-improve-hamming-oob-layout.patch
patching file target/linux/bcm63xx/patches-5.4/022-v5.8-mtd-rawnand-brcmnand-correctly-verify-erased-pages.patch
Reversed (or previously applied) patch detected!  Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file target/linux/bcm63xx/patches-5.4/022-v5.8-mtd-rawnand-brcmnand-correctly-verify-erased-pages.patch.rej
patching file target/linux/bcm63xx/patches-5.4/024-v5.8-mtd-rawnand-brcmnand-fix-CS0-layout.patch
Reversed (or previously applied) patch detected!  Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file target/linux/bcm63xx/patches-5.4/024-v5.8-mtd-rawnand-brcmnand-fix-CS0-layout.patch.rej
patching file target/linux/generic/backport-5.4/070-v5.5-MIPS-BPF-Restore-MIPS32-cBPF-JIT.patch
patching file target/linux/generic/backport-5.4/310-v5.6-mips-vdso-fix-jalr-t9-crash-in-vdso-code.patch
patching file target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch
patching file target/linux/generic/backport-5.4/716-v5.4-net-sfp-move-fwnode-parsing-into-sfp-bus-layer.patch
patching file target/linux/generic/backport-5.4/717-v5.5-net-sfp-rework-upstream-interface.patch
patching file target/linux/generic/backport-5.4/718-v5.5-net-sfp-fix-sfp_bus_put-kernel-documentation.patch
patching file target/linux/generic/backport-5.4/742-v5.5-net-sfp-add-support-for-module-quirks.patch
patching file target/linux/generic/backport-5.4/743-v5.5-net-sfp-add-some-quirks-for-GPON-modules.patch
patching file target/linux/generic/hack-5.4/221-module_exports.patch
patching file target/linux/generic/hack-5.4/301-mips_image_cmdline_hack.patch
patching file target/linux/generic/hack-5.4/641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch
Reversed (or previously applied) patch detected!  Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file target/linux/generic/hack-5.4/641-sch_cake-fix-IP-protocol-handling-in-the-presence-of.patch.rej
patching file target/linux/generic/hack-5.4/904-debloat_dma_buf.patch
patching file target/linux/generic/pending-5.4/300-mips_expose_boot_raw.patch
patching file target/linux/generic/pending-5.4/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
patching file target/linux/generic/pending-5.4/810-pci_disable_common_quirks.patch
patching file target/linux/lantiq/patches-5.4/0152-lantiq-VPE.patch
patching file target/linux/layerscape/patches-5.4/701-net-0231-enetc-Use-DT-protocol-information-to-set-up-the-port.patch
patching file target/linux/layerscape/patches-5.4/701-net-0234-enetc-WA-for-MDIO-register-access-issue.patch
patching file target/linux/layerscape/patches-5.4/701-net-0236-enetc-Replace-enetc_gregs-with-a-readers-writer-lock.patch
patching file target/linux/layerscape/patches-5.4/701-net-0237-enetc-Remove-mdio-bus-on-PF-probe-error-path.patch
patching file target/linux/layerscape/patches-5.4/701-net-0337-enetc-add-support-tsn-capabilities-qbv-qci-qbu-cbs.patch
patching file target/linux/layerscape/patches-5.4/806-dma-0019-dmaengine-fsl-edma-calculate-the-real-count-for-slav.patch
patching file target/linux/layerscape/patches-5.4/806-dma-0028-dmaengine-fsl-edma-Add-eDMA-support-for-QorIQ-LS1028.patch
patching file target/linux/layerscape/patches-5.4/811-kvm-0002-arm-arm64-KVM-allow-specifying-s2-prot-bits-when-map.patch
patching file target/linux/layerscape/patches-5.4/811-kvm-0003-arm-arm64-KVM-drop-qman-mmio-cacheable-mapping-hack.patch
patching file target/linux/layerscape/patches-5.4/812-pcie-0005-MLK-20684-PCI-Disable-MSI-on-CYW4356-and-CYW4359-chi.patch
patching file target/linux/layerscape/patches-5.4/812-pcie-0007-PCI-Disable-MSI-on-marvel-88w9098-and-88w8997-chips.patch
patching file target/linux/mediatek/patches-5.4/0400-eth-fix-rx-vlan-hw-offload.patch
patching file target/linux/mediatek/patches-5.4/0401-eth-fix-eth1-tx-timeout.patch
patching file target/linux/mediatek/patches-5.4/0402-net-ethernet-mtk_eth_soc-Always-call-mtk_gmac0_rgmii.patch
Reversed (or previously applied) patch detected!  Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file target/linux/mediatek/patches-5.4/0402-net-ethernet-mtk_eth_soc-Always-call-mtk_gmac0_rgmii.patch.rej
patching file target/linux/mediatek/patches-5.4/0991-dt-bindings-PCI-Mediatek-Update-PCIe-binding.patch
patching file target/linux/mediatek/patches-5.4/0999-hnat.patch
patching file target/linux/mediatek/patches-5.4/1000-eth-gdm-config-backport.patch
patching file target/linux/mediatek/patches-5.4/1001-eth-mtk-disable-TBDDONE-support.patch
patching file target/linux/mvebu/patches-5.4/012-net-mvneta-rely-on-build_skb-in-mvneta_rx_swbm-poll-.patch
patching file target/linux/mvebu/patches-5.4/013-net-mvneta-add-basic-XDP-support.patch
patching file target/linux/mvebu/patches-5.4/023-net-mvneta-fix-XDP-support-if-sw-bm-is-used-as-fallb.patch
patching file target/linux/mvebu/patches-5.4/561-mvebu-armada-38x-enable-libata-leds.patch
patching file target/linux/octeontx/patches-5.4/0004-PCI-add-quirk-for-Gateworks-PLX-PEX860x-switch-with-.patch
patching file target/linux/ramips/patches-5.4/0069-awake-rt305x-dwc2-controller.patch
patching file target/linux/ramips/patches-5.4/0400-net-ethernet-mediatek-use-napi_consume_skb.patch
patching file target/linux/ramips/patches-5.4/401-net-ethernet-mediatek-support-net-labels.patch
patching file target/linux/rockchip/patches-5.4/101-dts-rockchip-add-usb3-controller-node-for-RK3328-SoCs.patch
patching file target/linux/rockchip/patches-5.4/102-rockchip-enable-LAN-port-on-NanoPi-R2S.patch
patching file target/linux/rockchip/patches-5.4/103-arm64-rockchip-add-OF-node-for-USB-eth-on-NanoPi-R2S.patch
```

**Comment by adschm** [comment]
Just checked it locally, the patch from the staging tree seems to be improperly formatted. If you take the same commit from GitHub, it applies: https://github.com/adschm/openwrt/commit/fc6c9e82bd47adf6827b7d9ea7b9fcb4c067c998.patch

Of course, pulling the tree and cherry-picking also works.

**Comment by graysky2** [comment]
@adschm - How odd!  I did not realize you had a github mirror.  The patch from there applied cleanly.  Building now.  Can you advise on my previous question:

> @adschm - I will pull. build, and test on my hardware. Is your proposal for me to just rebase and force push to this PR or did you intent to just request the merge from your branch? Thanks.

**Comment by adschm** [comment]
> @adschm - I will pull. build, and test on my hardware. Is your proposal for me to just rebase and force push to this PR or did you intent to just request the merge from your branch? Thanks.

If you confirm that my commit is okay and my local build-tests run fine, I'd apply my version to master.

**Comment by graysky2** [comment]
@adschm - Built and booted into to now/no dmesg regressions.  Everything appears to be working, including SQM.

**Comment by graysky2** [comment]
@adschm - Since you've done all this work, would you mind taking a draft against [5.4.59-rc1](https://mirrors.edge.kernel.org/pub/linux/kernel/v5.x/stable-review/patch-5.4.59-rc1.xz)?  I am finding several patches therein to be rather difficult.

File: target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (curtdept) [reply target/linux/generic/backport-5.4/395-v5.8-net-sch_cake-Take-advantage-of-skb-hash-where-appropriate.patch:4]: This does seem to be upstreamed in master but I don't see it in the 5.4 branch yet.
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/net/sched?id=b0c19ed6088ab41dd2a727b60594b7297c15d6ce

**Comment by adschm** [comment]
> @adschm - Since you've done all this work, would you mind taking a draft against [5.4.59-rc1](https://mirrors.edge.kernel.org/pub/linux/kernel/v5.x/stable-review/patch-5.4.59-rc1.xz)? I am finding several patches therein to be rather difficult.

Actually, my main motivation was to get this off the table for now, as it included several kernel revisions already. I'm typically not dealing with kernel refreshs.

**Comment by adschm** [comment]
Just pushed my version to master. So, at least you can start clean with 5.4.59 now ...

**Comment by graysky2** [comment]
Thanks for the time you invested with this, @adschm!

**Comment by adschm** [comment]
@graysky2 I've started playing with 5.4.59 due to curiosity and solved some issues on the way, so maybe wait until I post a PR here to prevent duplicate work.

**Pull request closed without merging**